node/src/blockchain/blockchain_bridge.rs

```diff
@@ -1582,27 +1508,33 @@ mod tests {
         System::current().stop();
         system.run();
         let after = SystemTime::now();
-        let retrieve_transactions_params = retrieve_transactions_params_arc.lock().unwrap();
-        assert_eq!(
-            *retrieve_transactions_params,
-            vec![(
-                BlockNumber::Number(6u64.into()),
-                10006u64,
-                earning_wallet.address()
-            )]
-        );
+        let expected_transactions = RetrievedBlockchainTransactions {
+            new_start_block: 6040060u64,
+            transactions: vec![
+                BlockchainTransaction {
+                    block_number: 6040059,
+                    from: make_wallet("first_wallet"), // Points to topics of 1
+                    wei_amount: 42,                    // Its points to the field data
```

Comment:
Is this better?

```rust
        let expected_transactions = RetrievedBlockchainTransactions {
            new_start_block: 6040060u64,
            transactions: vec![
                BlockchainTransaction {
                    block_number: 6040059,
                    from: make_wallet("first_wallet"), // Relates to RPC response topics of 1
                    wei_amount: 42,                            // Relates to RPC response field data
                },
                BlockchainTransaction {
                    block_number: 6040060,
                    from: make_wallet("second_wallet"), // Relates to RPC response topics of 1
                    wei_amount: 55,                             // Relates to RPC response field data
                },
            ],
        };
```

Reply:
No, I'd need the comment to be more explicit about what is "topics of 1" and I'm not even sure about the rest. Which "field data"? Why is it important. 

It's so confusing. Above all, the comment probably has no or barely any value. I mean, maybe even if stated in better words. 

I can only judge what I can see on the screen and even though I give it an extra effort I must conclude the comment probably makes sense only to the person who wrote it and nobody else. Which is a bit irritating, because then no comment would be better than this.

Reply:
The `topics` are coming from the RPC response that was created in MBCS. I think these comments would make a lot more sense if we looked at the full test.

Reply:
I get that part still the comment can be confusing. Okay, I think, now that I'm sure we're talking about the same thing. What about this:

```
        let expected_transactions = RetrievedBlockchainTransactions {
            new_start_block: 6040060u64,
            transactions: vec![
                BlockchainTransaction {
                    block_number: 6040059,
                    // Wallet represented in the RPC response by the second 'topic' as: 0x000000000000000000000000000000000000000066697273745f77616c6c6574  
                    from: make_wallet("first_wallet"), 
                    wei_amount: 42,                         
                },
                BlockchainTransaction {
                    block_number: 6040060,
                    // Wallet represented in the RPC response by the second 'topic' as: 0x000000000000000000000000000000000000007365636f6e645f77616c6c6574  
                    from: make_wallet("second_wallet"),
                    wei_amount: 55,                             
                },
            ],
        };
```
This is a more explaining comment, I think. 
 
Just please check if I linked the right hex format to the wallet creating function it actually belongs to. :pray: